xcodeproj/internal/top_level_targets.bzl

```diff
@@ -475,3 +479,35 @@ def process_top_level_target(
             lldb_context = lldb_context,
         ),
     )
+
+def _get_app_icon_name(ctx, automatic_target_info):
+    """Attempts to find the applicaiton icon name.
+
+    Args:
+        ctx: The aspect context.
+        automatic_target_info: The `XcodeProjAutomaticTargetProcessingInfo` for
+            `target`.
+
+    Returns:
+        The application icon name, if found. Otherwise, None.
+    """
+    if automatic_target_info.app_icons == None:
+        return None
+
+    app_icons = getattr(ctx.rule.attr, automatic_target_info.app_icons, None)
+    if app_icons == None:
+        return None
+
+    for target in app_icons:
+        for file in target.files.to_list():
+            suffix_idx = file.short_path.find(".appiconset")
```

Comment:
Do we need to support the other bundle types as well, such as `.brandassets` (tvOS)?

Reply:
I added `.brandassets` support.